server/src/main/java/org/opensearch/action/admin/indices/shrink/TransportResizeAction.java

```diff
@@ -368,4 +373,24 @@ protected static int calculateTargetIndexShardsNum(
     protected String getClusterManagerActionName(DiscoveryNode node) {
         return super.getClusterManagerActionName(node);
     }
+
+    private static void validateClusterModeSettings(
+        final ResizeType type,
+        IndexMetadata sourceIndexMetadata,
+        ClusterSettings clusterSettings
+    ) {
+        boolean isMixed = clusterSettings.get(RemoteStoreNodeService.REMOTE_STORE_COMPATIBILITY_MODE_SETTING)
+            .equals(RemoteStoreNodeService.CompatibilityMode.MIXED);
+        boolean isRemoteStoreMigrationDirection = clusterSettings.get(RemoteStoreNodeService.MIGRATION_DIRECTION_SETTING)
+            .equals(RemoteStoreNodeService.Direction.REMOTE_STORE);
+        boolean isRemoteStoreEnabled = sourceIndexMetadata.getSettings().getAsBoolean(SETTING_REMOTE_STORE_ENABLED, false);
+        if (isMixed && isRemoteStoreMigrationDirection && !isRemoteStoreEnabled) {
```

Comment:
lets use `isRemoteStoreEnabled == false` as it is the convention . Rename variable to `idxRemoteStoreEnabled` .

Reply:
We also need to modify the logic here .

1. If the direction is remote and idx is remote,  allow 
2. If the direction is docrep and idx is non-remote , allow